examples/01_dirty_categories.py

```diff
@@ -1,274 +1,289 @@
 """
-Dirty categories: machine learning with non normalized strings
-==============================================================
+dirty_cat: Tackling Non-Normalized Categories In Machine Learning Settings
+==========================================================================
 
-Including strings that represent categories often calls for much data
-preparation. In particular categories may appear with many morphological
-variants, when they have been manually input or assembled from diverse
-sources.
+Handling strings that represent categories often requires substantial data
+preparation, especially when they contain various morphological variants
+emanating from manual input or diverse data sources.
 
-Here we look at a dataset on wages [#]_ where the column 'Employee
-Position Title' contains dirty categories. On such a column, standard
-categorical encodings leads to very high dimensions and can lose
-information on which categories are similar.
+..note:
+    A variable is considered categorical when it can take on one of a
+    limited, and usually fixed (discrete) number of possible values.
+    It is opposed to continuous variables and free text,
+    which can take an infinite number of unique values.
 
-We investigate various encodings of this dirty column for the machine
-learning workflow, predicting the 'Current Annual Salary' with gradient
-boosted trees. First we manually assemble a complex encoder for the full
-dataframe, after which we show a much simpler way, albeit with less fine
-control.
+Typically, we encounter several types of morphological variants in categories:
+- **Typos**: "France" vs "Frqnce"
+- **Abbreviations**: "USA" vs "United States of America"
+- **Duplicated values**: "France/FR"
+- **Alternate values**: "Bordeaux, FR" & "Bordeaux"
+- **Variations**: "avocat" & "avocate" (masculine & feminine of lawyer in French)
 
+While some of these variations carry meaning, others do not. For instance,
+"Frqnce" is simply misspelled, while "avocat" and "avocate" represent the same
+profession, performed by individuals of different genders. In a salary prediction
+context, the latter variation is likely significant for model accuracy, while the
+former is irrelevant.
 
-.. [#] https://www.openml.org/d/42125
+Using the `employee salaries <https://www.openml.org/d/42125>`_ dataset,
+we aim to predict wages.
 
+This notebook offers an overview of *dirty_cat*, a library designed to handle
+dirty categorical data in machine learning settings.
+We will explore the following topics:
 
- .. |TV| replace::
-     :class:`~dirty_cat.TableVectorizer`
+1. Comparing the performance of a traditional encoder, the |OneHotEncoder|,
+   with the encoders provided by the *dirty_cat* library
+2. Introducing a simpler approach to assembling a machine-learning pipeline
+3. Demonstrating how to join and augment dirty tables effectively
```

Comment:
I worry that the scope of this example is too big. Maybe I have the wrong feeling, but I'd like to be to say: one page describes one idea.

I see different ideas here, that I believe should probably be in different pages:
- Encoders: representing data, including in messy data
- Representing strings even when non-normalized
- Augmenting data (that definitely another track).

Reply:
I agree that this example should be focusing on point 1 and 2 noted here, which are the features of the "early" versions of dirty-cat.

It is not that easy to link all the features of the package in one example, as they have also different logic (e.g. encoding vs deduplicating).

We have other examples for 3 (fuzzy joining) and 4 (deduplicating)